### PR TITLE
Do not require exclusive bind of udp sockets

### PR DESCRIPTION
**Pull request by sammirata**
Check return value of setsockopt function call (SO_REUSEADDR)

### CODE REVIEW

**Comment by sammirata** [comment]
I signed it!

File: packager/media/file/udp_file_posix.cc [hunk 1]
```diff
@@ -187,6 +187,9 @@ bool UdpFile::Open() {
   local_sock_addr.sin_family = AF_INET;
   local_sock_addr.sin_port = htons(dest_port);
   local_sock_addr.sin_addr.s_addr  = htonl(dest_addr);
+  // We do not need to require exclusive bind of udp sockets
+  int optval = 1;
```
Comment (kqyang) [inline packager/media/file/udp_file_posix.cc]: Replace "int" with "const int" to make it clear that it is a const input parameter.

File: packager/media/file/udp_file_posix.cc [hunk 2]
```diff
@@ -187,6 +187,9 @@ bool UdpFile::Open() {
...
+  setsockopt(new_socket.get(), SOL_SOCKET, SO_REUSEADDR, &optval, sizeof(optval));
```
Comment (kqyang) [inline packager/media/file/udp_file_posix.cc]: Please check the return value of setsockopt. Log an error and return false if it fails.

I am not familiar with SO_REUSEADDR, just for my own knowledge, does it matter if we call setsockopt before or after bind?

File: packager/media/file/udp_file_posix.cc (hunk 1 above)
Comment (sammirata) [inline packager/media/file/udp_file_posix.cc]: Done

File: packager/media/file/udp_file_posix.cc (hunk 2 above)
Comment (sammirata) [inline packager/media/file/udp_file_posix.cc]: Done ... You need to set SO_REUSEADDR before calling bind or it will fail

**Comment by sammirata** [comment]
I sent a new pull request with the additional requested changes

File: AUTHORS [hunk 3]
```diff
@@ -18,3 +18,4 @@ Google Inc. <*@google.com>
 Leandro Moreira <leandro.ribeiro.moreira@gmail.com>
 Philo Inc. <*@philo.com>
 The Chromium Authors <*@chromium.org>
+Sergio Ammirata <sergio@ammirata.net>
```
Comment (kqyang) [inline AUTHORS]: Please order in alphabetical order. Same for contributors.

File: packager/media/file/udp_file_posix.cc [hunk 4]
```diff
@@ -188,8 +188,12 @@ bool UdpFile::Open() {
   local_sock_addr.sin_port = htons(dest_port);
   local_sock_addr.sin_addr.s_addr  = htonl(dest_addr);
   // We do not need to require exclusive bind of udp sockets
-  int optval = 1;
-  setsockopt(new_socket.get(), SOL_SOCKET, SO_REUSEADDR, &optval, sizeof(optval));
+  const int optval = 1;
+  if (setsockopt(new_socket.get(), SOL_SOCKET, SO_REUSEADDR, &optval, sizeof(optval)) < 0)
```
Comment (kqyang) [inline packager/media/file/udp_file_posix.cc]: We have a 80-characters limit per line; also, "{" should not be in its own line. Can you change it to:

```
  if (setsockopt(new_socket.get(), SOL_SOCKET, SO_REUSEADDR, &optval,
                 sizeof(optval)) < 0) {
    LOG(ERROR)
        << "Could not apply the SO_REUSEADDR property to the UDP socket.";
    return false;
  }
```

File: AUTHORS (hunk 3 above)
Comment (sammirata) [inline AUTHORS]: Done

File: packager/media/file/udp_file_posix.cc (hunk 4 above)
Comment (sammirata) [inline packager/media/file/udp_file_posix.cc]: Done

**Pull request merged**